contracts/ERC721A.sol

```diff
@@ -375,15 +375,24 @@ contract ERC721A is Context, ERC165, IERC721, IERC721Metadata {
             _ownerships[startTokenId].startTimestamp = uint64(block.timestamp);
 
             uint256 updatedIndex = startTokenId;
+            uint256 end = updatedIndex + quantity;
 
-            for (uint256 i; i < quantity; i++) {
-                emit Transfer(address(0), to, updatedIndex);
-                if (safe && !_checkOnERC721Received(address(0), to, updatedIndex, _data)) {
-                    revert TransferToNonERC721ReceiverImplementer();
+            if (safe && to.isContract()) {
+                while (updatedIndex != end) {
```

Comment:
The `!=` check is more gas optimized and also doubles as an overflow guard.

Even if the user sets the `_startTokenId()` to a value near `2**256`, it will still work.

If there is overflow, the loop will loop so many times that it gives an out of gas error.